io-engine/src/lvs/lvs_snapshot.rs

```diff
@@ -0,0 +1,74 @@
+use super::Error;
+use crate::lvs::{LogicalVolume, Lvol, Lvs};
+use async_trait::async_trait;
+use spdk_rs::libspdk::spdk_blob_is_read_only;
+
+pub struct LvolSnapshot {
+    logical_vol: Lvol,
+    group_id: Option<String>,
+    entity_id: Option<String>,
+    parent: Option<String>,
+}
+
+#[async_trait(? Send)]
+impl LogicalVolume for LvolSnapshot {
+    /// returns the name of the Snapshot
+    fn name(&self) -> String {
+        self.logical_vol.as_bdev().name().to_string()
+    }
+
+    /// returns the UUID of the Snapshot
+    fn uuid(&self) -> String {
+        self.logical_vol.as_bdev().uuid_as_string()
+    }
+    /// return lvs for the Logical Volume
+    fn lvs(&self) -> Lvs {
+        Lvs::from_inner_ptr(self.logical_vol.as_inner_ref().lvol_store)
+    }
+
+    /// returns the pool name of the Snapshot
+    fn pool_name(&self) -> String {
+        self.logical_vol.lvs().name().to_string()
+    }
+
+    /// returns the pool uuid of the Snapshot
+    fn pool_uuid(&self) -> String {
+        self.logical_vol.lvs().uuid()
+    }
+
+    ///  wipe the first 8MB if unmap is not supported on failure the operation
+    /// needs to be repeated
+    async fn wipe_super(&self) -> Result<(), Error> {
```

Comment:
Snapshots are R/O entities, so they should not allow modifying underlying blocks.
It's better to move this method to Volume-specific operations.